test/test.py

```diff
@@ -7,160 +7,230 @@
+
+    def __enter__(self):
+        torchaudio.set_audio_backend(self.new_backend)
+        return self.new_backend
+
+    def __exit__(self, type, value, traceback):
+        backend = self.previous_backend
+        torchaudio.set_audio_backend(backend)
+
+
 class Test_LoadSave(unittest.TestCase):
     test_dirpath, test_dir = common_utils.create_temp_assets_dir()
     test_filepath = os.path.join(test_dirpath, "assets",
                                  "steam-train-whistle-daniel_simon.mp3")
 
     def test_1_save(self):
-        # load signal
-        x, sr = torchaudio.load(self.test_filepath, normalization=False)
-
-        # check save
-        new_filepath = os.path.join(self.test_dirpath, "test.wav")
-        torchaudio.save(new_filepath, x, sr)
-        self.assertTrue(os.path.isfile(new_filepath))
-        os.unlink(new_filepath)
-
-        # check automatic normalization
-        x /= 1 << 31
-        torchaudio.save(new_filepath, x, sr)
-        self.assertTrue(os.path.isfile(new_filepath))
-        os.unlink(new_filepath)
-
-        # test save 1d tensor
-        x = x[0, :]  # get mono signal
-        x.squeeze_()  # remove channel dim
-        torchaudio.save(new_filepath, x, sr)
-        self.assertTrue(os.path.isfile(new_filepath))
-        os.unlink(new_filepath)
-
-        # don't allow invalid sizes as inputs
-        with self.assertRaises(ValueError):
-            x.unsqueeze_(1)  # L x C not C x L
-            torchaudio.save(new_filepath, x, sr)
-
-        with self.assertRaises(ValueError):
-            x.squeeze_()
-            x.unsqueeze_(1)
-            x.unsqueeze_(0)  # 1 x L x 1
-            torchaudio.save(new_filepath, x, sr)
-
-        # don't save to folders that don't exist
-        with self.assertRaises(OSError):
-            new_filepath = os.path.join(self.test_dirpath, "no-path",
-                                        "test.wav")
-            torchaudio.save(new_filepath, x, sr)
-
-        # save created file
-        sinewave_filepath = os.path.join(self.test_dirpath, "assets",
-                                         "sinewave.wav")
-        sr = 16000
-        freq = 440
-        volume = 0.3
-
-        y = (torch.cos(
-            2 * math.pi * torch.arange(0, 4 * sr).float() * freq / sr))
-        y.unsqueeze_(0)
-        # y is between -1 and 1, so must scale
-        y = (y * volume * (2**31)).long()
-        torchaudio.save(sinewave_filepath, y, sr)
-        self.assertTrue(os.path.isfile(sinewave_filepath))
-
-        # test precision
-        new_precision = 32
-        new_filepath = os.path.join(self.test_dirpath, "test.wav")
-        si, ei = torchaudio.info(sinewave_filepath)
-        torchaudio.save(new_filepath, y, sr, new_precision)
-        si32, ei32 = torchaudio.info(new_filepath)
-        self.assertEqual(si.precision, 16)
-        self.assertEqual(si32.precision, new_precision)
-        os.unlink(new_filepath)
+        for backend in ["sox"]:
```

Comment:
You don't have to do this. You can call a function within a score. Prefix this with an underscore and then write another test that calls it into within different scopes.